src/app/vendors/Vendors.tsx

```diff
@@ -96,10 +87,14 @@ function Vendors({
 }: Props) {
   const defs = useD2Definitions();
   const isPhonePortrait = useIsPhonePortrait();
-  const [characterId, setCharacterId] = useState<string>();
   const [filterToUnacquired, setFilterToUnacquired] = useState(false);
 
-  const selectedStoreId = characterId || getCurrentStore(stores)?.id;
+  // once the page is loaded, user can select this. on initial load, default to current char.
+  const [selectedStoreId, setSelectedStoreId] = useState(getCurrentStore(stores)?.id);
+
+  let vendorGroups = useSelector((state: RootState) =>
+    vendorGroupsForCharacterSelector(state, selectedStoreId)
+  );
```

Comment:
Something I've started doing is currying the selector, e.g.:

```
const vendorGroupsForCharacterSelector = (selectedStoreId: string) => (state: RootState) => ...;

// Later...
const vendorGroups = useSelector(vendorGroupsForCharacterSelector(selectedStoreId));
```

That may be hard to do given that this is generated from reselect though.

Reply:
yeah reselect needs re-reselect or some really ugly syntax to establish different caches for dynamically created selectors